packages/dataviews/src/components/dataviews-filters/filter.tsx

```diff
@@ -494,10 +494,21 @@ export default function Filter( {
 			return filterInView?.value?.includes( element.value );
 		} );
 	} else if ( filterInView?.value !== undefined ) {
+		const field = fields.find( ( f ) => f.id === filter.field );
+		let label = filterInView.value;
+
+		if ( field?.type === 'datetime' && typeof label === 'string' ) {
+			try {
+				label = new Date( label ).toLocaleString();
```

Comment:
If label is null it will return the epoch (Jan 1, 1970), not sure if we want to handle this case as we are using ts and label is not expected to be null.